### ISSUE TITLE
add a Duration::toDateInterval and Duration::fromDateInterval methods

### ISSUE THREAD

**Issue by nyamsprod** [delight-im/PHP-Temporal#4]
For interoperability you should also add the following methods to the Duration class:
- `Duration::toDateInterval` 
- `Duration::fromDateInterval`

**Comment by ocram** [comment]
Definitely, thank you!

For `toDateInterval`, one solution would be to call [`Duration#toIso8601`](https://github.com/delight-im/PHP-Temporal/blob/a32148d3e353d0428d235ef264b734cce73d03e3/src/Duration.php#L151) and pass the resulting string to the constructor of the `DateInterval` class.

For `fromDateInterval`, do you think it should just read the properties from a `DateInterval` instance and pass them to [`Duration::fromDateTime`](https://github.com/delight-im/PHP-Temporal/blob/a32148d3e353d0428d235ef264b734cce73d03e3/src/Duration.php#L309)? We should probably pay special attention to the `days` property on the `DateInterval` instances that is set only when the instances are the result of a diff.

**Comment by nyamsprod** [comment]
yes indeed the `days` property is the one to pay attention to. :)
